app/models/region.rb

```diff
@@ -134,24 +134,35 @@ def dashboard_analytics(period:, prev_periods:, include_current_period: true)
   def syncable_patients
     case region_type
       when "block"
-        registered_patients.with_discarded
-          .or(assigned_patients.with_discarded)
-          .union(appointed_patients.with_discarded)
+        Patient.with_discarded.where(id: syncable_patient_ids)
       else
         registered_patients.with_discarded
     end
   end
 
-  def registered_patients
-    Patient.where(registration_facility: facility_regions.pluck(:source_id))
+  def syncable_patient_ids(exclude_facility: nil)
+    case region_type
+      when "block"
+        registered_patients(exclude_facility: exclude_facility)
+          .with_discarded.select(:id)
+          .or(assigned_patients(exclude_facility: exclude_facility).with_discarded.select(:id))
+          .union(appointed_patient_ids(exclude_facility: exclude_facility).with_discarded)
+          .pluck(:id)
```

Comment:
Having both `select` and `pluck` is a bit unintuitive, shall we leave a comment here explaining the rationale?

Reply:
No need to pluck, fixed in a0edbc5ce3ed820302f5c2934f74688de9edc445.